src/js/forms.js

```diff
@@ -0,0 +1,127 @@
+class Forms {
+
+	constructor (FormEl, opts) {
+		this.FormEl = FormEl;
+		this.validFormEls = [
+			HTMLFormElement,
+			HTMLInputElement,
+			HTMLSelectElement,
+			HTMLButtonElement,
+			HTMLTextAreaElement
+		];
+
+		const declaredTestEvent = this.FormEl.getAttribute('data-o-forms-test-event');
```

Comment:
Test event is the name I've given to the event on which to run the validation checks - explained in the readme, either `blur` or `submit`. It's a terrible name, happy to take suggestions...